polaris-core/src/main/java/org/apache/polaris/core/config/FeatureConfiguration.java

```diff
@@ -234,4 +233,36 @@ public static void enforceFeatureEnabledOrThrow(
           .description("If true, the policy-store endpoints are enabled")
           .defaultValue(true)
           .buildFeatureConfiguration();
+
+  public static final FeatureConfiguration<Boolean> ALLOW_SPECIFYING_FILE_IO_IMPL =
+      PolarisConfiguration.<Boolean>builder()
+          .key("ALLOW_SPECIFYING_FILE_IO_IMPL")
+          .description(
+              "Config key for whether to allow setting the FILE_IO_IMPL using catalog properties. "
+                  + "Must only be enabled in dev/test environments, should not be in production systems.")
+          .defaultValue(false)
+          .buildFeatureConfiguration();
+
+  public static final FeatureConfiguration<Boolean> ALLOW_INSECURE_STORAGE_TYPES =
```

Comment:
Not sure when my comment got dropped but this still looks redundant with the existing config

Reply:
I informs users who may not be aware of code-level intricacies that certain storage types are considered insecure. Without it, a user may naively assume that adding something to the list of storage types at run time is "good enough". That may be true for some types, but not for others.

Reply:
I believe we can probably improve the granularity of this check, but I'd suggest to do that in a follow-up PR.

Reply:
To me, `ALLOW_INSECURE_STORAGE_TYPES = false` seems to imply that Polaris is somehow guaranteeing that no insecure File IO implementations are allowed, which I don't think is something the service can guarantee after this change.

Reply:
> I think this property is good enough as it's generic to any storage type (not only FILE).

Is that true? What qualifies a storage type as insecure?

If the answer is "the storage type can use HadoopFileIO", then all 4 storage types are insecure.